apollo-runtime-kotlin/src/commonMain/kotlin/com/apollographql/apollo/interceptor/ApolloRequest.kt

```diff
@@ -4,13 +4,97 @@ import com.apollographql.apollo.api.ApolloExperimental
 import com.apollographql.apollo.api.ExecutionContext
 import com.apollographql.apollo.api.Operation
 import com.apollographql.apollo.api.CustomScalarAdapters
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Query
+import com.apollographql.apollo.api.Subscription
+import com.benasher44.uuid.Uuid
 import com.benasher44.uuid.uuid4
 
 @ApolloExperimental
-class ApolloRequest<D : Operation.Data>(
-    val operation: Operation<D>,
-    val customScalarAdapters: CustomScalarAdapters,
-    val executionContext: ExecutionContext
-) {
-  val requestUuid = uuid4()
+interface ApolloRequest<D : Operation.Data> {
```

Comment:
curious if we should make it a sealed class?

Reply:
Makes sense 👍